app/models/agents/webhook_agent.rb

```diff
@@ -95,6 +99,10 @@ def validate_options
       unless options['secret'].present?
         errors.add(:base, "Must specify a secret for 'Authenticating' requests")
       end
+
+      unless !(options['code'].presence && options['code'].blank?)
```

Comment:
We don't want to make it mandatory, so I'd do:

`if options['code'].present? && options['code'].to_s !~ /\A\s*(\d+|\{.*)\s*\z/`

(We should probably start doing something like this more generally.  This says that it must be present, and must either be all digits, or start with `{`, which would be the start of a Liquid template.)